lib/node-utils.js

```diff
@@ -244,6 +244,32 @@ function isComma(token) {
     return token.kind === SyntaxKind.CommaToken;
 }
 
+/**
+ * Returns true if the given TSToken is a comment
+ * @param  {TSToken}  token the TypeScript token
+ * @returns {boolean}       is commment
+ */
+function isComment(token) {
+    return (token.kind === SyntaxKind.SingleLineCommentTrivia || token.kind === SyntaxKind.MultiLineCommentTrivia) || (token.kind >= SyntaxKind.JSDocTypeExpression && token.kind <= SyntaxKind.JSDocTypeLiteral);
+}
+
+/**
+ * Returns true if the given TSToken is inside a comment
+ * Non-comment type tokens are generated for types in JSDoc Blocks
+ * @param  {TSToken}  token the TypeScript token
+ * @returns {boolean}       is inside a commment
+ */
+function isInsideComment(token) {
+    if (token.kind === SyntaxKind.SourceFile) {
```

Comment:
Is this check sufficient? I was wondering if the algorithm could alternatively be:

Reply:
We could also avoid crawling up the tree on every node if we knew exactly what kind of tokens can exist in JSDoc comments (scoping the check to those node types). Maybe someone with more knowledge of could help with that?